File: src/main/java/com/github/tetrisanalyzer/piece/PieceZ.java
package com.github.tetrisanalyzer.piece;

public class PieceZ extends Piece {
    @Override public byte number() { return 2; }
    @Override public char character() { return 'Z'; }
    @Override protected int[] widths() { return new int[] { 3, 2 }; }
    @Override protected int[] heights() { return new int[] { 2, 3 }; }
    @Override protected PieceShape[] shapes() {
        return new PieceShape[] {
            new PieceShape(new Point(0,0), new Point(1,0), new Point(1,1), new Point(2,1)),
            new PieceShape(new Point(1,0), new Point(0,1), new Point(1,1), new Point(0,2))
        };
    }
}

File: langs/scala/src/main/scala/nu/tengstrand/tetrisanalyzer/piece/PieceZ.scala
package nu.tengstrand.tetrisanalyzer.piece

object PieceZ {
  def apply() = new PieceZ
}

class PieceZ extends Piece {
  val number = 2.toByte
  val character = 'Z'
  protected val widths = Array(3, 2)
  protected val heights = Array(2, 3)
  protected val shapes = Array(
    new PieceShape(Array(Point(0,0), Point(1,0), Point(1,1), Point(2,1))),
    new PieceShape(Array(Point(1,0), Point(0,1), Point(1,1), Point(0,2)))
  )
}
